### ISSUE TITLE
"(forked without exec)" recording does not replay

### ISSUE THREAD

**Issue by kellerb** [mozilla/rr#1466]
When told to replay a forked without exec recording, `rr replay -p <pid>`, rr processes for awhile and returns without launching gdb.

Running on current master, 7bf7fd443888ad7c5ee8f6120f95d2c719c3b9ae

**Comment by rocallahan** [comment]
If you do `rr replay -p` with a pid that never did an `execve`, you'll get that behavior, and that's more or less by design, since `-p` is specified to stop at the first exec. So what do you think we should do differently here? We could exit immediately with an error explaining that that pid never execs (we know that based on the summary data collected for `rr ps`), or we could be more DWIM and quietly treat it as `-f` when there's no exec. I prefer the former.

**Comment by kellerb** [comment]
Oops, I created this issue while operating under the incorrect assumption that `rr replay -p <pid>` just meant "start the replay identified by `<pid>`". After your comment and actually reading the help, I see that the solution to my problem is "use `-f` instead of `-p`".

I retract my issue with the behavior of `-p` on "forked without exec" processes. Though, if I were to choose between your two options, I would also prefer the "exit immediately with error" route.

Semi-relatedly, while debugging a different process created via fork-exec, I used `-f` to start the replay before the exec call, and while stepping into the exec call, hit an rr assertion failure. Should I be able to step through the exec call in the debugger and end up debugging the "new" binary? If so, I'll create a new issue for this.
